gateway/src/apicast/configuration_loader/remote_v2.lua

```diff
@@ -162,6 +146,18 @@ local function reset_env()
          resty_env.value('APICAST_CONFIGURATION_LOADER') == 'boot'
 end
 
+-- Returns a table that represents paths and query parameters for the current endpoint:
+-- http://${THREESCALE_PORTAL_ENDPOINT}/<env>.json?host=host
+-- http://${THREESCALE_PORTAL_ENDPOINT}/admin/api/account/proxy_configs/<env>.json?host=host&version=version
+local function configuration_endpoint_params(env, host, portal_endpoint_path)
+  local version = resty_env.get(
+      format('APICAST_SERVICE_%s_CONFIGURATION_VERSION', service_id)
```

Comment:
where is this `service_id` coming from?

Reply:
apparently, that environment variable (for configuring a specific service version) only works when specific conditions are met:

* THREESCALE_PORTAL_ENDPOINT points to an admin portal, and
* APICAST_LOAD_SERVICES_WHEN_NEEDED is **not** configured

Because in order to fetch a specific version it's necessary to load services individually (so that the version can be specified for each service configuration URL). And this is not possible via master or the new services endpoint. This is currently the behavior we have also in master and even in previous stable versions.

For now, I updated the logic to fall back to the "old" endpoint and load services one by one when `APICAST_SERVICE_%s_CONFIGURATION_VERSION` is configured (and `THREESCALE_PORTAL_ENDPOINT` is an admin portal), and I added logs to notify when services will be loaded this way, individually. This means we would preserve the current behaviour when the version env var is set.

We need to verify and confirm that this is the best way to handle things.

Reply:
I added the same "exceptions" for `APICAST_SERVICES_LIST` and `APICAST_SERVICES_FILTER_BY_URL` env vars.